controllers/contributorController.js

```diff
@@ -130,29 +106,61 @@ const controller = {
         .map((item) => item.trim())
         .filter((item) => validSkills.includes(item));
 
-      const newContributor = await ContributorModel.create({
-        title,
-        project_id: project._id,
-        skills: skillsArr,
-        is_remote,
-        description,
-        commitment_level,
-        available_slots,
-        city,
-        remuneration,
-      });
+      data.skills = skillsArr;
+      data.project_id = req.projectID;
+
+      const newContributor = await ContributorModel.create(data);
 
       return res.status(201).json(newContributor);
     } catch (error) {
+      console.log(`error: ${error}`);
       return res.status(500).json({
         error: 'Failed to fetch data',
       });
     }
-
   },
 
   update: async (req, res) => {
+    let data = null;
+    try {
+      data = await validator.details.validateAsync(req.body);
+    } catch (error) {
+      console.log(error);
+      return res.status(400).json({
+        error: 'Invalid input',
+      });
+    }
+
+    const relations = await RelationshipModel.find({ contributor_id: req.contributorID });
+
+    const skillsArr = data.skills
+      .split(',')
+      .map((item) => item.trim())
+      .filter((item) => validSkills.includes(item));
+    data.skills = skillsArr;
```

Comment:
Why do you need to do all this? Can't you just map the skills to an array?

Reply:
I think when the form data comes in, it will be in String format?
I am not sure how to submit form data in array format hence right now assumed it's String and need to process it like this

Reply:
Ok